packages/x-data-grid-pro/src/hooks/features/dataSource/useGridDataSourceCache.ts

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import useLazyRef from '@mui/utils/useLazyRef';
+import { useGridApiMethod } from '@mui/x-data-grid';
+import { GridPrivateApiPro } from '../../../models/gridApiPro';
+import { DataGridProProcessedProps } from '../../../models/dataGridProProps';
+import { GridGetRowsParams, GridGetRowsResponse, GridDataSourceCache } from '../../../models';
+import { GridDataSourceCacheApi } from './interfaces';
+
+class SimpleServerSideCache {
```

Comment:
While this simple cache is enough for demo purposes, we should provide a more advanced cache implementation required for real-world use cases – at least with configurable cache time.
What do you think?

Reply:
I wanted to make the default cache as simple as possible, but I agree `ttl` could be added, as it's one of the most basic features of a cache. Added!
Let me know if it feels better now.